viite-UI/src/view/ProjectEditForm.js

```diff
@@ -436,7 +436,13 @@
         } else if (objectDropdown_1) {
           projectCollection.saveCutProjectLinks(projectCollection.getTmpDirty(), objectDropdown_0.value, objectDropdown_1.value);
         } else {
-          projectCollection.saveProjectLinks(projectCollection.getTmpDirty(), objectDropdown_0.value);
+            const endDistance = $('#endDistance') [0];
+            var changedValue;
+            var touchedEndDistance = false;
+            if (endDistance) changedValue = Number(endDistance.value);
+            if (!isNaN(changedValue) && !isNaN(parseInt(endDistanceOriginalValue)) && changedValue !== endDistanceOriginalValue)
```

Comment:
Mites jos on pelkästään !isNaN(changedValue). Eiks sillonkii pitäis olla
touchedEndDistance = true;
eli vaikka originaalissa oiski jotain epälukuhöpöä?

Reply:
Tai siis oikeastaan, silloin, kun on
 !isNaN(changedValue) && /*ei-huutomerkkiä*/isNaN(parseInt(endDistanceOriginalValue))

Reply:
Eikös tossa vaan testata että ne on numeroita ja sitten verrata että changedValue on eri kuin endDistanceOriginalValue

Voisi kirjoittaa uudestaan käyttäen typeof:

if (typeof changedValue === 'number' && typeof endDistanceOriginalValue === 'number' && changedValue !== endDistanceOriginalValue)

Reply:
Joo, se on varmasti logiikkalauseen pointti.
Tartuin siis siihen, että jos joiden isNan-arvo täytyy testata, on oletettavaa, että niissä voi olla jotai ei-numeraalista, ja ilmeisesti toisistaan riippumatta.
Nyt jos changedValuessa on järkevä numero, ei arvo on muutettu, mutta endDistanceoriginalValuessa ei ole järkevää arvoa, niin mutta iffin sisältöä ei mennä suorittamaan, vaikka arvoa on muutettu.